addons/hr_timesheet/hr_timesheet_view.xml

```diff
@@ -51,23 +104,39 @@
                     <field name="project_id"/>
                     <field name="task_id"/>
                     <field name="name"/>
-                    <field name="department_id"/>
+                    <filter name="mine" string="My Timesheets" domain="[('user_id', '=', uid)]"/>
+                    <separator/>
+                    <filter name="week" string="This Week"
+                            domain="[
+                                '&amp;',
+                                ('date', '>=', (context_today() + relativedelta(weeks=-1,days=1,weekday=0)).strftime('%Y-%m-%d')),
+                                ('date', '&lt;=', (context_today() + relativedelta(weekday=6)).strftime('%Y-%m-%d')),
+                            ]"/>
+                    <filter name="month" string="This Month"
+                            domain="[
+                                '&amp;',
+                                ('date', '>=', (context_today() + relativedelta(day=1)).strftime('%Y-%m-%d')),
+                                ('date', '&lt;=', (context_today() + relativedelta(months=1, day=1, days=-1)).strftime('%Y-%m-%d')),
```

Comment:
> That's strange to have both day and days arguments. 

it's the only way to get the last day of the month: go to next month (months=1), the first day of the month (day=1) and go one day back (days=-1).

In relativedelta, the "s" suffix means relative delta and no "s" means absolute so "day=1" means "the first day (of the month)" while "days=1" means "next day".

The application order is defined as absolute then relative so it works nicely for what we need.